src/sage/arith/misc.py

```diff
@@ -3708,18 +3739,31 @@ def CRT_vectors(X, moduli):
 
         sage: CRT_vectors([vector(ZZ, [2,3,1]), Sequence([1,7,8], ZZ)], [8,9])          # needs sage.modules
         [10, 43, 17]
+
+    ``CRT_vectors`` also works for some non-coprime moduli::
+
+        sage: CRT_vectors([[6],[0]],[10, 4])
+        [16]
+        sage: CRT_vectors([[6],[0]],[10, 10])
+        Traceback (most recent call last):
+        ...
+        ValueError: solution does not exist
     """
     # First find the CRT basis:
     if not X or len(X[0]) == 0:
         return []
     n = len(X)
     if n != len(moduli):
         raise ValueError("number of moduli must equal length of X")
-    a = CRT_basis(moduli)
+    res = CRT_basis(moduli, require_coprime_moduli=False)
+    a = res[0]
     modulus = prod(moduli)
-    return [sum(a[i] * X[i][j] for i in range(n)) % modulus
-            for j in range(len(X[0]))]
-
+    candidate = [sum(a[i] * X[i][j] for i in range(n)) % modulus
+                 for j in range(len(X[0]))]
+    if not res[1] and any(X[i][j] != candidate[j] % moduli[i] for i in range(n)
+       for j in range(len(X[i]))):
+        raise ValueError("solution does not exist")
```

Comment:
```suggestion
        for j in range(len(X[i]))):
            raise ValueError("solution does not exist")
```
Python's standard indentation is 4 spaces per level.